client/src/middleware.ts

```diff
@@ -17,19 +17,26 @@ function isAuthenticated(request: NextRequest): boolean {
 }
 
 export function middleware(request: NextRequest) {
-  const { pathname } = request.nextUrl;
+  const { pathname, search } = request.nextUrl;
   if (isAuthenticated(request)) {
     if (pathname.startsWith("/auth")) {
       return NextResponse.redirect(new URL("/user-boards", request.url));
     }
   } else {
-    if ((pathname.startsWith("/user-boards") || pathname.startsWith("/board")) && !pathname.startsWith("/auth")) {
-      return NextResponse.redirect(new URL("/auth/login", request.url));
+    if (pathname.startsWith("/user-boards") && !pathname.startsWith("/auth")) {
```

Comment:
If `pathname` starts with `/user-boards`, can it start with `/auth` at the same time?

Reply:
oh yes, done